packages/backend-common/src/reading/GitlabUrlReader.ts

```diff
@@ -117,13 +118,24 @@ export class GitlabUrlReader implements UrlReader {
     const { etag, signal } = options ?? {};
     const { ref, full_name, filepath } = parseGitUrl(url);
 
+    let repoFullName = full_name;
+
+    const relativePath = getGitLabIntegrationRelativePath(
+      this.integration.config,
+    );
+
+    // Considering self hosted gitlab with relative
+    if (relativePath) {
+      repoFullName = trimStart(full_name, relativePath);
```

Comment:
@daydy16 nice catch!
I misunderstood how loadash's trimStrart works as I didn't realize the second parameter is a list of characters, I think a simple `.replace()` should do the trick and match the `relativePath` only as it will stop after the first match as done [here](https://github.com/backstage/backstage/pull/12403/files#diff-46e51f9a59e02a0645b31f2510402a31699feac5f3ca8de5117794fba93056adR158), I think I have time for a quick patch this to PR later this week.

Reply:
@daydy16 -  made some changes to fix this, could you verify this is working for you?
https://github.com/OpenSourceZombie/backstage/tree/ahmed/fix-edge-case

Reply:
> 
Sorry, forgot to publish my review.

Reply:
> I fixed it with removing the / first, and then trimming it from the full_name:

Sorry, forgot to publish my review.

Reply:
@OpenSourceZombie will you raise a PR for the changes too so we can get them shipped in the mainline release on Tuesday?

Reply:
Yes.